sdk/javascript/src/utils/arrayHelpers.js

```diff
@@ -0,0 +1,82 @@
+const { BufferView } = require('./BufferView');
+
+const readArrayImpl = (bufferInput, FactoryClass, accessor, shouldContinue) => {
+	const view = new BufferView(bufferInput);
+	const elements = [];
+	let previousElement = null;
+	let i = 0;
+	while (shouldContinue(i, view)) {
+		const element = FactoryClass.deserialize(view.buffer);
+
+		if (0 >= element.size)
+			throw RangeError('element size has invalid size');
```

Comment:
is this possible?

Reply:
it could happen on malformed data and we do not want infinite loop, right? 😬